version: "3.9"

services:
  bot:
    build:
      context: .
    networks:
      - bot_network
    restart: always
    environment:
      PG_HOST: postgres
      PG_PORT: 5432
      PG_USER: python-problems-bot
      PG_PASSWORD: python-problems-bot  # change on production
      PG_DATABASE: python-problems-bot

      MONGO_HOST: mongo
      MONGO_USER: python-problems-bot
      MONGO_PASSWORD: python-problems-bot

      TOKEN: 6691015922:AAFNzx2yZD6MOp233WGNSBYYyz2VVumV-Dg  # change on production
    depends_on:
      postgres:
        condition: service_started
      mongo:
        condition: service_started
      migrate:
        condition: service_completed_successfully

  migrate:
    image: migrate/migrate
    networks:
      - bot_network
    volumes:
      - ./migrations:/migrations
    command: [
      "-path",
      "/migrations",
      "-database",
      "postgres://python-problems-bot:python-problems-bot@postgres:5432/python-problems-bot?sslmode=disable",
      "up"
    ]
    depends_on:
      postgres:
        condition: service_healthy

  postgres:
    image: postgres:15.0-alpine
    networks:
      - bot_network
    volumes:
      - postgres-data:/var/lib/postgresql/data
    ports:
      - "5436:5432"
    environment:
      POSTGRES_USER: python-problems-bot
      POSTGRES_PASSWORD: python-problems-bot # change on production
      POSTGRES_DB: python-problems-bot
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U python-problems-bot" ]
      interval: 5s
      timeout: 5s
      retries: 3

  mongo:
#    image: mvertes/alpine-mongo
    image: mongo
    networks:
      - bot_network
    volumes:
      - mongo-data:/data/db
    environment:
      MONGO_INITDB_ROOT_USERNAME: python-problems-bot
      MONGO_INITDB_ROOT_PASSWORD: python-problems-bot

networks:
    bot_network:

volumes:
  postgres-data:
  mongo-data:
